http/vertx-web-validation/src/main/java/io/quarkus/ts/vertx/web/validation/ValidationHandlerOnRoutes.java

```diff
@@ -0,0 +1,96 @@
+package io.quarkus.ts.vertx.web.validation;
+
+import static io.vertx.ext.web.validation.builder.Parameters.param;
+import static io.vertx.json.schema.common.dsl.Schemas.numberSchema;
+import static io.vertx.json.schema.common.dsl.Schemas.stringSchema;
+import static io.vertx.json.schema.draft7.dsl.Keywords.maximum;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.quarkus.vertx.web.Route;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.validation.RequestParameters;
+import io.vertx.ext.web.validation.ValidationHandler;
+import io.vertx.ext.web.validation.builder.ValidationHandlerBuilder;
+import io.vertx.json.schema.SchemaParser;
+import io.vertx.json.schema.SchemaRouter;
+import io.vertx.json.schema.SchemaRouterOptions;
+
+@ApplicationScoped
+public class ValidationHandlerOnRoutes {
+    //TODO when Quarkus use vert.x version 4.4.6 we can use SchemaRepository instead of SchemaParser with SchemaRouter
+    //private SchemaRepository schemaRepository =SchemaRepository.create(new JsonSchemaOptions().setDraft(Draft.DRAFT7).setBaseUri(BASEURI));
+    private SchemaParser schemaParser;
+    private SchemaRouter schemaRouter;
+
+    @Inject
+    Vertx vertx;
+
+    private Router router;
+
+    @Inject
+    ShopResource shopResource;
+
+    @PostConstruct
+    void initialize() {
+        router = Router.router(vertx);
```

Comment:
I don't think users should typically do that. You can do `@Observes Router router` and add route on application path router.

Reply:
Ah, you do that little below, in which case I don't comprehend why you want to create your own router? Please elaborate.

Reply:
Okay, I think you do that to test Reactive Route. IMO you can drop that completely.